src/bridge.ts

```diff
@@ -65,13 +66,21 @@ class DenoBridge {
     // a malformed semver range. If this does happen, let's throw an error so
     // that the tests catch it.
     if (downloadedVersion === undefined) {
-      throw new Error('Could not read downloaded binary')
+      const error = new Error(
+        'There was a problem setting up the Edge Functions environment. To try a manual installation, visit https://ntl.fyi/install-deno.',
+      )
+
+      if (this.onAfterDownload) {
+        this.onAfterDownload(error)
+      }
```

Comment:
[dust] This is highly dusty :) but we could write this as suggested, also in all the other places, which might help to bring down the amount of lines in this file enough to not have to disable eslint `max-lines` on line 213. Though I think it is better to do this in a follow up where we just do this, instead of adding more changes to this PR.

```suggestion
      this.onAfterDownload?.(error)
```